lib/rubocop/ast/node/mixin/method_dispatch_node.rb

```diff
@@ -211,7 +211,7 @@ def lambda?
       #
       # @return [Boolean] whether this method is a lambda literal
       def lambda_literal?
-        block_literal? && loc.expression && loc.expression.source == '->'
+        loc.expression && loc.expression.source == '->' && block_literal?
```

Comment:
are there nodes that don't have `.loc.expression`?

Reply:
Seems like no. Thank you, removed that check.